tests/unit/events/queue_tests.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+import json
+import unittest
+
+from baseplate.events import (
+    Event,
+    EventError,
+    EventQueue,
+    EventQueueFullError,
+    EventTooLargeError,
+    MAX_EVENT_SIZE,
+)
+from baseplate.message_queue import MessageQueue, TimedOutError
+
+from ... import mock
+
+
+class EventTests(unittest.TestCase):
+    @mock.patch("uuid.uuid4")
+    @mock.patch("time.time")
+    def test_minimal_constructor(self, time, uuid):
+        time.return_value = 333
+        uuid.return_value = "1-2-3-4"
+
+        event = Event("topic", "type")
+
+        self.assertEqual(event.topic, "topic")
+        self.assertEqual(event.event_type, "type")
+        self.assertEqual(event.timestamp, 333000)
+        self.assertEqual(event.id, "1-2-3-4")
+
+    def test_timestamp_specified(self):
+        timestamp = datetime.datetime(2016, 2, 23, 0, 3, 17)
+        event = Event("topic", "type", timestamp=timestamp)
+        self.assertEqual(event.timestamp, 1456185797000)
+
```

Comment:
Ah, hmm. I guess the way to do it without adding `pytz` or another timezone library would be your own class that inherits from `datetime.tzinfo`. It looks like it would probably just take something like this:

```
class FakeTZ(datetime.tzinfo):
    def utcoffset(self, dt):
        return datetime.timedelta(hours=2)
```

Not too bad, might be worth adding since an error that starts allowing non-UTC timestamps could cause a pretty big mess in the data.